Src/Support/Google.Apis/Requests/ClientServiceRequest.cs

```diff
@@ -379,19 +379,17 @@ private void AddParameters(RequestBuilder requestBuilder, ParameterCollection in
         {
             foreach (var parameter in inputParameters)
             {
-                IParameter parameterDefinition;
-
-                if (!RequestParameters.TryGetValue(parameter.Key, out parameterDefinition))
+                if (!RequestParameters.TryGetValue(parameter.Key, out IParameter parameterDefinition))
                 {
                     throw new GoogleApiException(Service.Name,
                         String.Format("Invalid parameter \"{0}\" was specified", parameter.Key));
                 }
 
                 string value = parameter.Value;
-                if (!ParameterValidator.ValidateParameter(parameterDefinition, value))
+                if (!ParameterValidator.ValidateParameter(parameterDefinition, value, out string error))
                 {
                     throw new GoogleApiException(Service.Name,
-                        string.Format("Parameter validation failed for \"{0}\"", parameterDefinition.Name));
+                        string.Format("Parameter validation failed for \"{0}\" : {1}", parameterDefinition.Name, error));
```

Comment:
I'd use an interpolated string literal here as well. (We've been gradually moving to them, generally as we edit code that uses string.Format. Feel free to either change or leave the call a few lines up.)